app/controllers/ally_controller.rb

```diff
@@ -0,0 +1,142 @@
+class AllyController < ApplicationController
+
+  #
+  # REST methods
+  #
+
+  ##
+  # Returns whether the plugin is enabled for the account. If its
+  # enabled extra information such as the client ID and base URL
+  # of the Ally API is returned as well
+  def enabled
+    plugin_data = get_plugin_data
+    data = {
+      :enabled => plugin_data[:enabled],
+      :clientId => plugin_data[:client_id],
+      :baseUrl => plugin_data[:base_url]
+    }
+    render :json => data
+  end
+
+  ##
+  # Sign a request to the Ally REST API but don't execute it.
+  # The UI will execute the request
+  def sign
+    if (request = get_request())
+      plugin_data = get_plugin_data()
+      puts request.to_yaml
+      data = {
+        :clientId => plugin_data[:client_id],
+        :baseUrl => plugin_data[:base_url],
+        :path => request.path,
+        :header => request.to_hash["authorization"][0],
+        :body => request.body,
+      }
+      render :json => data
+    end
+  end
+
+  ##
+  # Proxy data from the Ally REST API
+  def proxy
+    if (request = get_request())
+      # TODO: Cache the HTTP object per hostname allowing
+      # for connection pooling
+      plugin_data = get_plugin_data()
+      uri = URI.parse(plugin_data[:base_url])
+      http = Net::HTTP.new(uri.host, uri.port)
+      response = http.request(request)
+
+      # Render successful requests as JSON
+      if response.code == 200 || response.code == 201
+        render :json => response.body, :status => response.code
+      else
+        render :text => response.body, :status => response.code
+      end
+    end
+  end
+
+  protected
+
+  # Caches HTTP objects PER ally hostname. This allows for HTTP
+  # connection pooling which avoids opening a new connection to
+  # the Ally REST API
+  http_caches = {}
+
+  ##
+  # Get a request object that can be constructed for the parameters
+  # to the `sign` or `proxy` endpoints
+  def get_request()
+    # Respond with a 400 if the Ally plugin hasn't been enabled
+    plugin_data = get_plugin_data
+    if !plugin_data[:enabled]
+      render :text => "Ally has not been enabled yet", :status => :bad_request
+      return false
+    end
+
+    # Send an appropriate response depending on the role of the user
+    role = get_course_role()
+    if role == nil
```

Comment:
I think the guidelines avoid explicit comparison to nil in favour of using `role.nil?`